tests/Unit/ProxyClient/VarnishTest.php

```diff
@@ -99,6 +99,43 @@ public function testBanPathEmptyHost()
         $varnish->banPath('/articles/.*', 'text/html', $hosts);
     }
 
+    public function testTagsHeaders()
+    {
+        $varnish = new Varnish(['127.0.0.1:123'], 'fos.lo', $this->client);
+        $varnish->setDefaultBanHeaders(
+            ['A' => 'B']
+        );
+        $varnish->setDefaultBanHeader('Test', '.*');
+        $varnish->invalidateTags(['post-1', 'post-type-3'])->flush();
+
+        $requests = $this->getRequests();
+
+        $this->assertCount(1, $requests);
+        $this->assertEquals('BAN', $requests[0]->getMethod());
+
+        $this->assertEquals('(post\-1|post\-type\-3)(,.+)?$', $requests[0]->getHeaderLine('X-Cache-Tags'));
+        $this->assertEquals('fos.lo', $requests[0]->getHeaderLine('Host'));
+
+        // This being taken into account, might not be expected from user POV?
```

Comment:
That tag invilaidation, now separate from ban, takes the extra ban options into account because of code reuse.